internal/controller/standalone_pgadmin/users_test.go

```diff
@@ -0,0 +1,599 @@
+	})
+
+	// We only test v7 because if we did v8 then the writePGAdminUsers would
+	// be called and that method has its own tests later in this file
+	t.Run("PodHealthyVersionNotSet", func(t *testing.T) {
+		pgadmin := pgadmin.DeepCopy()
+		pod := pod.DeepCopy()
+
+		pod.DeletionTimestamp = nil
+		pod.Status.ContainerStatuses =
+			[]corev1.ContainerStatus{{Name: naming.ContainerPGAdmin}}
+		pod.Status.ContainerStatuses[0].State.Running =
+			new(corev1.ContainerStateRunning)
+		pod.Status.ContainerStatuses[0].ImageID = "fakeSHA"
+
+		r := new(PGAdminReconciler)
+		r.Client = fake.NewClientBuilder().WithObjects(pod).Build()
+
+		calls := 0
+		r.PodExec = func(
+			namespace, pod, container string,
+			stdin io.Reader, stdout, stderr io.Writer, command ...string,
+		) error {
+			calls++
+
+			assert.Equal(t, pod, "pgadmin-123-0")
+			assert.Equal(t, namespace, pgadmin.Namespace)
+			assert.Equal(t, container, naming.ContainerPGAdmin)
+
+			// Simulate a v7 version of pgAdmin by setting stdout to "7" for
+			// podexec call in reconcilePGAdminMajorVersion
+			stdout.Write([]byte("7"))
+			return nil
+		}
+
+		assert.NilError(t, r.reconcilePGAdminUsers(ctx, pgadmin))
+		assert.Equal(t, calls, 1, "PodExec should be called once")
+		assert.Equal(t, pgadmin.Status.MajorVersion, 7)
+		assert.Equal(t, pgadmin.Status.ImageSHA, "fakeSHA")
+	})
+
+	t.Run("PodHealthyShaChanged", func(t *testing.T) {
+		pgadmin := pgadmin.DeepCopy()
+		pgadmin.Status.MajorVersion = 7
+		pgadmin.Status.ImageSHA = "fakeSHA"
+		pod := pod.DeepCopy()
+
+		pod.DeletionTimestamp = nil
+		pod.Status.ContainerStatuses =
+			[]corev1.ContainerStatus{{Name: naming.ContainerPGAdmin}}
+		pod.Status.ContainerStatuses[0].State.Running =
+			new(corev1.ContainerStateRunning)
+		pod.Status.ContainerStatuses[0].ImageID = "newFakeSHA"
+
+		r := new(PGAdminReconciler)
+		r.Client = fake.NewClientBuilder().WithObjects(pod).Build()
+
+		calls := 0
+		r.PodExec = func(
+			namespace, pod, container string,
+			stdin io.Reader, stdout, stderr io.Writer, command ...string,
+		) error {
+			calls++
+
+			// Simulate a v7 version of pgAdmin by setting stdout to "7" for
+			// podexec call in reconcilePGAdminMajorVersion
+			stdout.Write([]byte("7"))
+			return nil
+		}
+
+		assert.NilError(t, r.reconcilePGAdminUsers(ctx, pgadmin))
+		assert.Equal(t, calls, 1, "PodExec should be called once")
+		assert.Equal(t, pgadmin.Status.MajorVersion, 7)
+		assert.Equal(t, pgadmin.Status.ImageSHA, "newFakeSHA")
+	})
+}
+
+func TestReconcilePGAdminMajorVersion(t *testing.T) {
+	ctx := context.Background()
+	pod := corev1.Pod{}
+	pod.Namespace = "test-namespace"
+	pod.Name = "pgadmin-123-0"
+	reconciler := &PGAdminReconciler{}
+
+	podExecutor := func(
+		_ context.Context, stdin io.Reader, stdout, stderr io.Writer, command ...string,
+	) error {
+		return reconciler.PodExec(pod.Namespace, pod.Name, "pgadmin", stdin, stdout, stderr, command...)
+	}
+
+	t.Run("SuccessfulRetrieval", func(t *testing.T) {
```

Comment:
maybe a test for `FailedRetrieval` would be worthwhile? Where "python" returns an unexpected value